spec/facter/resolvers/windows/networking_spec.rb

```diff
@@ -250,5 +250,52 @@
         expect(resolver.resolve(:domain)).to eql(domain)
       end
     end
+
+    context 'when it successfully retrieved networking information but the interface has non-UTF-8 friendly name' do
+      let(:error_code) { NetworkingFFI::ERROR_SUCCES }
+      let(:adapter) do
+        OpenStruct.new(OperStatus: NetworkingFFI::IF_OPER_STATUS_UP, IfType: NetworkingFFI::IF_TYPE_ETHERNET_CSMACD,
+                       DnsSuffix: dns_ptr, FriendlyName: friendly_name_ptr, Flags: 0, Mtu: 1500,
+                       FirstUnicastAddress: ptr, Next: ptr, to_ptr: FFI::Pointer::NULL)
+      end
+      let(:ptr) { instance_spy(FFI::Pointer) }
+      let(:dns_ptr) { instance_spy(FFI::Pointer) }
+      let(:friendly_name_ptr) { instance_spy(FFI::Pointer) }
```

Comment:
Based on https://stackoverflow.com/questions/57541170/utf16-big-endian-to-utf8-conversion-for-failed-for-0xdcf0, the two byte sequence `0xDC`, `0xF0` is not a valid unicode code point. You could set the Friendly Name to that and verify facter correctly replaces it with the replacement character `\uFFFD`. And rather then stubbing out the FFI pointer, just create one that points to the desired byte array. Note since we're using UTF-16LE (little endian) the byte array needs to be reversed below:

```ruby
let(:friendly_name_ptr) do
  bytes = "\xF0\xDC".force_encoding(Encoding::UTF_16LE).bytes
  ptr = FFI::MemoryPointer.new(:uchar, bytes.length)
  ptr.write_array_of_uchar(bytes)
  ptr
```
Then in the test when you call `resolver.resolve(:interfaces)`, the network resolver will call `adapter_address[:FriendlyName]`, which will return the `friendly_name_ptr` above and call `read_wide_string_without_length(replace_invalid_chars: true)` on it. Then in your test you can verify that:

```ruby
expect(resolved_interface.keys.first).to eql("\uFFFD")
```